modules/videoio/src/cap_ffmpeg_impl.hpp

```diff
@@ -1598,7 +1602,12 @@ static int icv_av_write_frame_FFMPEG( AVFormatContext * oc, AVStream * video_st,
 #endif
     int ret = OPENCV_NO_FRAMES_WRITTEN_CODE;
 
+#if LIBAVFORMAT_BUILD >= CALC_FFMPEG_VERSION(57, 0, 0)
+    if (video_st->codec->codec_type == AVMEDIA_TYPE_VIDEO &&
+        video_st->codec->codec_id == AV_CODEC_ID_RAWVIDEO) {
+#else
     if (oc->oformat->flags & AVFMT_RAWPICTURE) {
+#endif
```

Comment:
`AVFMT_RAWPICTURE` have been deprecated/eliminated completely (there is no replacement for this), so code block under this "if" is not used at all anymore.

Please disable these code block completely:
```
#if LIBAVFORMAT_BUILD < CALC_FFMPEG_VERSION(57, 0, 0)
    if (oc->oformat->flags & AVFMT_RAWPICTURE) {
        /* raw video case. The API will change slightly in the near
           futur for that */
        AVPacket pkt;
        av_init_packet(&pkt);

#ifndef PKT_FLAG_KEY
#define PKT_FLAG_KEY AV_PKT_FLAG_KEY
#endif

        pkt.flags |= PKT_FLAG_KEY;
        pkt.stream_index= video_st->index;
        pkt.data= (uint8_t *)picture;
        pkt.size= sizeof(AVPicture);

        ret = av_write_frame(oc, &pkt);
    } else
#endif
    {
        /* encode the image */
        AVPacket pkt;
        av_init_packet(&pkt);
        ...
    }
```

Please apply similar changes for other `AVFMT_RAWPICTURE` entries (condition is negative).
```
#if LIBAVFORMAT_BUILD < CALC_FFMPEG_VERSION(57, 0, 0)
    if (!(oc->oformat->flags & AVFMT_RAWPICTURE))
#endif
    {
        ... code ...
    }
```